gsl/span

```diff
@@ -670,6 +670,19 @@ span<typename Ptr::element_type>
 make_span(Ptr& cont)
 { return span<typename Ptr::element_type>(cont); }
 
+template <class T, class = std::enable_if_t<std::is_pod<T>::value>>
+span<const byte, sizeof(T)>
+as_bytes(const T& e) noexcept
+{
+    return{ reinterpret_cast<const byte*>(&e),sizeof(T) };
+}
+
+template <class T, class = std::enable_if_t<std::is_trivially_copyable<T>::value>>
```

Comment:
@LYP951018: That would also be a possibility. I somewhat like your first suggestion better, because it is more explicit (in that we won't take temporaries as well as that the sparameter will not be modified inside the function). In any case, I'd first wait on what neil has to say about the if`, before we go to the how.